exporters/metric/cortex/cortex.go

```diff
@@ -62,3 +68,78 @@ func InstallNewPipeline(config Config, options ...push.Option) (*push.Controller
 	global.SetMeterProvider(pusher.Provider())
 	return pusher, nil
 }
+
+// AddHeaders adds required headers as well as all headers in Header map to a http request.
+func (e *Exporter) AddHeaders(req *http.Request) {
+	// Cortex expects Snappy-compressed protobuf messages. These two headers are hard-coded as they
+	// should be on every request.
+	req.Header.Add("Content-Encoding", "snappy")
+	req.Header.Set("Content-Type", "application/x-protobuf")
+
+	// Add all user-supplied headers to the request.
+	for name, field := range e.Config.Headers {
+		req.Header.Add(name, field)
+	}
+}
+
+// BuildRequest creates an http POST request with a []byte as the body and headers attached.
+func (e *Exporter) BuildRequest(message []byte) (*http.Request, error) {
+	// Create the request with the endpoint and message. The message should be a Snappy-compressed
+	// protobuf message.
+	req, err := http.NewRequest("POST", e.Config.Endpoint, bytes.NewBuffer(message))
```

Comment:
Minor: Since we already have a dependency on the `http` package we could also use `http.MethodPost` instead of `"POST"` strings.